unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ChatController/ChatController.cs

```diff
@@ -143,36 +143,45 @@ public void UpdateChannelMembers(string payload)
     [UsedImplicitly]
     public void UpdateChannelInfo(string payload)
     {
-        var msg = JsonUtility.FromJson<ChannelInfoPayload>(payload);
-        var channelId = msg.channelId;
-        var channel = new Channel(channelId, msg.unseenMessages, msg.memberCount, msg.joined, msg.muted, msg.description);
-        var justLeft = false;
+        var msg = JsonUtility.FromJson<ChannelInfoPayloads>(payload);
 
-        if (channels.ContainsKey(channelId))
+        foreach (var channelInfo in msg.channelInfoPayload)
         {
-            justLeft = channels[channelId].Joined && !channel.Joined;
-            channels[channelId].CopyFrom(channel);
-        }
-        else
-            channels[channelId] = channel;
+            var channelId = channelInfo.channelId;
+            var channel = new Channel(channelId, channelInfo.unseenMessages, channelInfo.memberCount, channelInfo.joined, channelInfo.muted, channelInfo.description);
+            var justLeft = false;
 
-        if (justLeft)
-        {
-            OnChannelLeft?.Invoke(channelId);
+            if (channels.ContainsKey(channelId))
+            {
+                justLeft = channels[channelId].Joined && !channel.Joined;
+                channels[channelId].CopyFrom(channel);
+            }
+            else
+                channels[channelId] = channel;
 
-            // TODO (responsibility issues): extract to another class
-            AudioScriptableObjects.leaveChannel.Play(true);
-        }
+            if (justLeft)
+            {
+                OnChannelLeft?.Invoke(channelId);
 
-        OnChannelUpdated?.Invoke(channel);
+                // TODO (responsibility issues): extract to another class
+                AudioScriptableObjects.leaveChannel.Play(true);
+            }
+
+            OnChannelUpdated?.Invoke(channel);
+        }
     }
 
     // called by kernel
     [UsedImplicitly]
     public void JoinChannelConfirmation(string payload)
     {
-        var msg = JsonUtility.FromJson<ChannelInfoPayload>(payload);
-        var channel = new Channel(msg.channelId, msg.unseenMessages, msg.memberCount, msg.joined, msg.muted, msg.description);
+        var msg = JsonUtility.FromJson<ChannelInfoPayloads>(payload);
+
+        if (msg.channelInfoPayload.Length == 0)
```

Comment:
the data structure is wrong, there should not be a field called `channelInfoPayload`

Reply:
confirmed: it is designed in this way from kernel 👍